### DISCUSSION TITLE
Push shopping list to Home Assistant

### DISCUSSION THREAD

**Discussion by YaMoef** [mealie-recipes/mealie#3154] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

A possibility to push/sync ingredients on the shopping list in Mealie to Home Assistant shopping lists (To-Do lists integration). This makes it easier to keep shopping lists centralized and attach on events of the shopping lists. Maybe it is possible to do this using a 3rd party service like node-red to poll the shopping lists and push it like that to home assistant. I have not researched this route since I have no experience with node-red

### Please provide a concise description of the feature that would resolve your issue.

I would like to see a button in the shopping list which can push all the items over to a defined shopping list of Home Assistant. The integration ideally can be set up in the settings. A plus would be that the shopping lists are synced bi directional. In this case the extra button can be removed.

### Please consider and list out some caveats or tradeoffs made in your design decision

A button has to be added. Some extra fields have to be stored into the database to set up the integration.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
At this time we're not planning on building/maintaining any native syncing functionality into Mealie, instead we're just supporting the ability to create your own integrations via the API. I built a separate [Unified Shopping List](https://github.com/michael-genson/Unified-Shopping-List) app that does what you're asking for with Mealie, Alexa, and Todoist. I'd be willing to look into implementing a Home Assistant sync there as well

**Reply by YaMoef** [reply]
Thanks for the suggestion about USL (and of course all the work and effort that went into it), but honestly I'm not a fan of a centralized app to integrate with Alexa since I don't need Alexa. I will look into a node-red flow first and come back after it

**Reply by michael-genson** [reply]
Yup, makes sense! Node Red should be able to handle it no problem, you'll want to leverage Mealie's apprise notifications and API to handle the Mealie side, then have some sort of workflow triggered by the HA shopping list that feeds changes back into Mealie

**Comment by PaulRitzkat** [comment]
Michael on the HA forum already did the work to create an automation [here](https://community.home-assistant.io/t/mealie-in-ha/603560/27) to sync the shoppinglist from mealie to HA
